cumulus/pallets/xcmp-queue/src/migration.rs

```diff
@@ -80,6 +60,89 @@ mod v1 {
 	}
 }
 
+mod v2 {
+	use super::*;
+
+	#[frame_support::storage_alias]
+	pub(crate) type QueueConfig<T: Config> = StorageValue<Pallet<T>, QueueConfigData, ValueQuery>;
+
+	#[derive(Copy, Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug, TypeInfo)]
+	pub struct QueueConfigData {
+		pub suspend_threshold: u32,
+		pub drop_threshold: u32,
+		pub resume_threshold: u32,
+		#[deprecated(note = "Will be removed")]
+		pub threshold_weight: Weight,
+		#[deprecated(note = "Will be removed")]
+		pub weight_restrict_decay: Weight,
+		#[deprecated(note = "Will be removed")]
+		pub xcmp_max_individual_weight: Weight,
+	}
+
+	impl Default for QueueConfigData {
+		fn default() -> Self {
+			#![allow(deprecated)]
+			Self {
+				drop_threshold: 48,    // 64KiB * 48 = 3MiB
+				suspend_threshold: 32, // 64KiB * 32 = 2MiB
+				resume_threshold: 8,   // 64KiB * 8 = 512KiB
+				// unused:
+				threshold_weight: Weight::from_parts(100_000, 0),
+				weight_restrict_decay: Weight::from_parts(2, 0),
+				xcmp_max_individual_weight: Weight::from_parts(
+					20u64 * WEIGHT_REF_TIME_PER_MILLIS,
+					DEFAULT_POV_SIZE,
+				),
+			}
+		}
+	}
+
+	/// Migrates `QueueConfigData` from v1 (using only reference time weights) to v2 (with
+	/// 2D weights).
+	pub struct MigrationToV2<T: Config>(PhantomData<T>);
+
+	impl<T: Config> OnRuntimeUpgrade for MigrationToV2<T> {
+		#[allow(deprecated)]
+		fn on_runtime_upgrade() -> Weight {
+			let translate = |pre: v1::QueueConfigData| -> v2::QueueConfigData {
+				v2::QueueConfigData {
+					suspend_threshold: pre.suspend_threshold,
+					drop_threshold: pre.drop_threshold,
+					resume_threshold: pre.resume_threshold,
+					threshold_weight: Weight::from_parts(pre.threshold_weight, 0),
+					weight_restrict_decay: Weight::from_parts(pre.weight_restrict_decay, 0),
+					xcmp_max_individual_weight: Weight::from_parts(
+						pre.xcmp_max_individual_weight,
+						DEFAULT_POV_SIZE,
+					),
+				}
+			};
+
+			if v2::QueueConfig::<T>::translate(|pre| pre.map(translate)).is_err() {
+				log::error!(
+					target: crate::LOG_TARGET,
+					"unexpected error when performing translation of the QueueConfig type \
+					during storage upgrade to v2"
+				);
+			}
+
+			T::DbWeight::get().reads_writes(1, 1)
+		}
+	}
+
+	/// [`MigrationToV2`] wrapped in a
+	/// [`VersionedMigration`](frame_support::migrations::VersionedMigration), ensuring the
+	/// migration is only performed when on-chain version is 1.
+	#[allow(dead_code)]
+	pub type VersionCheckedMigrationToV2<T> = frame_support::migrations::VersionedMigration<
```

Comment:
I think the docs mention it, but i think its a bad convention to name the versioned ones with `Versioned`. Rather we should name the unchecked ones with something like `Unchecked...`. WDYT @liamaharon ?

Reply:
Sounds good. Done.